diff --git a/examples/arc75/artifacts/ARC75.approval.teal b/examples/arc75/artifacts/ARC75.approval.teal
index bdcaa30b..e846a55b 100644
--- a/examples/arc75/artifacts/ARC75.approval.teal
+++ b/examples/arc75/artifacts/ARC75.approval.teal
@@ -337,11 +337,17 @@ setAppWhitelist:
 
 if1_else:
 	// examples/arc75/arc75.algo.ts:65
-	// this.verifyMBRPayment(this.txnGroup[this.txn.groupIndex - 1], preMBR)
+	// this.verifyMBRPayment(this.txnGroup[this.txn.groupIndex - 1] as PayTxn, preMBR)
 	frame_dig 0 // preMBR: uint64
 	txn GroupIndex
 	int 1
 	-
+	gtxns TypeEnum
+	int pay
+	assert
+	txn GroupIndex
+	int 1
+	-
 	callsub verifyMBRPayment
 
 if1_end:
